### PR TITLE
Create LightningKokkos v0.30.0 RC branch

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/tests_linux_x86_nvidia_gpu.yml [hunk 1]
```diff
@@ -36,8 +36,10 @@ jobs:
 
       - name: Install required packages
         run: |
+          sudo apt-get update && sudo apt-get -y -q install gcc-${{ env.GCC_VERSION }} g++-${{ env.GCC_VERSION }} gcovr lcov | apt --fix-broken install | \
           sudo apt-get update && sudo apt-get -y -q install gcc-${{ env.GCC_VERSION }} g++-${{ env.GCC_VERSION }} gcovr lcov
```
Comment (AmintorDusko) [inline .github/workflows/tests_linux_x86_nvidia_gpu.yml:40]: Should this line be removed?
```suggestion
```

File: pennylane_lightning_kokkos/_version.py [hunk 2]
```diff
@@ -17,4 +17,4 @@
 """
 
 
-__version__ = "0.30.0-dev1"
+__version__ = "0.30.0rc0"
```
Comment (AmintorDusko) [inline pennylane_lightning_kokkos/_version.py:20]: Probably not important, but we usually do:
```suggestion
__version__ = "0.30.0-rc0"
```

File: .github/workflows/tests_linux_x86_nvidia_gpu.yml (hunk 1 above)
Comment (mlxd) [reply .github/workflows/tests_linux_x86_nvidia_gpu.yml:40]: Gone in PR #61

File: pennylane_lightning_kokkos/_version.py (hunk 2 above)
Comment (mlxd) [reply pennylane_lightning_kokkos/_version.py:20]: Docs allow either, though no period/dash is reported often more than if it is included. Either way, I don't mind.

**Pull request merged**